Sources/PackageGraph/DependencyResolver.swift

```diff
@@ -919,13 +919,27 @@ public class DependencyResolver<
     ///                  constraints for the same container identifier.
     /// - Returns: A satisfying assignment of containers and versions.
     /// - Throws: DependencyResolverError, or errors from the underlying package provider.
-    func resolveAssignment(constraints: [Constraint]) throws -> AssignmentSet {
+    func resolveAssignment(constraints: [Constraint], pins: [Constraint] = []) throws -> AssignmentSet {
+
+        // Create a constraint set with the input pins.
+        var allConstraints = ConstraintSet()
+        for constraint in pins {
+            if let merged = allConstraints.merging(constraint) {
+                allConstraints = merged
+            } else {
+                // FIXME: We should issue a warning if the pins can't be merged
```

Comment:
Seems like this should be a precondition or fatal error or something, not just silent?

Reply:
People can modify resolved file by hand so I thought warning would be better.